File: cracking/isUniq.js
// str = 'hello' => false
// str = 'helo' => true
function isUniq(str) {
  let arr = str.split('').sort((a,b) => a - b);
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === arr[i+1]) {
      return false;
    }
  }
  return true;
}


console.log(isUniq('hello'));
console.log(isUniq('helo'));

File: cracking/arraystring/is_uniq.rb
# hello => false
# helo => true

def is_uniq(str)
  arr = str.chars.sort!
  arr.each_with_index do |el, i|
    if el == arr[i + 1]
      return false
    end
  end
  true
end

p is_uniq('hello')
p is_uniq('helo')
